src/SFA.DAS.EAS.Portal/Application/Services/AccountDocumentServiceWithDuplicateCheck.cs

```diff
@@ -19,6 +19,11 @@ public Task<AccountDocument> Get(long id, CancellationToken cancellationToken =
             return _accountDocumentService.Get(id, cancellationToken);
         }
 
+        public Task<AccountDocument> GetOrCreate(long id, CancellationToken cancellationToken = default)
```

Comment:
Would it be worth splitting the Get/GetOrCreate and the Save out into separate interfaces to stop duplication of code?

Reply:
The clients would have to inject both the interfaces though.

Reply:
But it could mean we could only inject one where necessary if we only needed to get or save to the repo. It's preference either way in my opinion

Reply:
Every client does a GetOrCreate and then a Save.

Reply:
Worth keeping as is then 👍